src/Nethermind/Nethermind.Core/IWriteBatch.cs

```diff
@@ -5,5 +5,8 @@
 
 namespace Nethermind.Core
 {
-    public interface IWriteBatch : IDisposable, IWriteOnlyKeyValueStore;
+    public interface IWriteBatch : IDisposable, IWriteOnlyKeyValueStore
+    {
+        void Clear();
```

Comment:
`Clear` is needed to reset data in case of an error in the middle of the transaction, when no data should actually be written. Current version with write on `Dispose` will submit it regardless.

I would prefer to make the batch follow explicit-commit approach, but that would affect a lot of code.